# 1.Import Library

In [71]:
#library yang digunakan
from urllib.request import urlopen
from bs4 import BeautifulSoup
from collections import deque
import ssl
ssl._create_default_https_context = ssl._create_unverified_context
import warnings
warnings.filterwarnings('ignore')

# 2.Mendapatakan Parameter Untuk Input Pada AJAX



In [72]:
#melakukan crawling untuk mendpatkan value dropdown yang nantinya akan digunakan sebagai input ajax
wiki_link = "https://bansm.kemdikbud.go.id/akreditasi"
html = urlopen(wiki_link).read()
soup = BeautifulSoup(html, 'html.parser')

###  Parameter Provinsi

In [73]:
#mendapatkan nama provinsi
prov = soup.find("select", {"id": "id_provinsi"}).findAll('option')
values = [o.get("value") for o in prov]
prov = values[1:]
prov
prov_name = soup.find("select", {"id": "id_provinsi"})
nama_prov = []
for value in prov_name.stripped_strings:
    nama_prov.append(value)
#     print(value)
nama_prov = nama_prov[1:]


In [74]:
nama_prov

['ACEH',
 'BALI',
 'BANTEN',
 'BENGKULU',
 'D.I. YOGYAKARTA',
 'D.K.I. JAKARTA',
 'GORONTALO',
 'JAMBI',
 'JAWA BARAT',
 'JAWA TENGAH',
 'JAWA TIMUR',
 'KALIMANTAN BARAT',
 'KALIMANTAN SELATAN',
 'KALIMANTAN TENGAH',
 'KALIMANTAN TIMUR',
 'KALIMANTAN UTARA',
 'KEPULAUAN BANGKA BELITUNG',
 'KEPULAUAN RIAU',
 'LAMPUNG',
 'LUAR NEGERI',
 'MALUKU',
 'MALUKU UTARA',
 'NUSA TENGGARA BARAT',
 'NUSA TENGGARA TIMUR',
 'PAPUA',
 'PAPUA BARAT',
 'RIAU',
 'SULAWESI BARAT',
 'SULAWESI SELATAN',
 'SULAWESI TENGAH',
 'SULAWESI TENGGARA',
 'SULAWESI UTARA',
 'SUMATERA BARAT',
 'SUMATERA SELATAN',
 'SUMATERA UTARA']

In [75]:
prov

['11',
 '51',
 '36',
 '17',
 '34',
 '31',
 '75',
 '15',
 '32',
 '33',
 '35',
 '61',
 '63',
 '62',
 '64',
 '65',
 '19',
 '21',
 '18',
 '9999',
 '81',
 '82',
 '52',
 '53',
 '94',
 '91',
 '14',
 '76',
 '73',
 '72',
 '74',
 '71',
 '13',
 '16',
 '12']

###  Mendapatkan Parameter Jenjang

In [76]:
#mendapatkan jenjang sekolah sebagai input nanti yang akan dilakukan
jenjang = soup.find("select", {"id": "bentuk"}).findAll('option')
values = [o.get("value") for o in jenjang]
jenjang = values[1:]
jenjang
jenjang_name = soup.find("select", {"id": "bentuk"})
nama_jenjang = []
for value in jenjang_name.stripped_strings:
    nama_jenjang.append(value)
#     print(value)
nama_jenjang = nama_jenjang[1:]


In [77]:
nama_jenjang

['SD/MI',
 'SMP/MTS',
 'SMK',
 'SMA/MA',
 'SLB',
 'SDLB',
 'SMPLB',
 'SMLB',
 'SPK SD',
 'SPK SMP',
 'SPK SMA']

In [78]:
jenjang

['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11']

### Mendapatkan Parameter Kabupaten 

In [79]:
from requests import Session
import pandas as pd
from bs4 import BeautifulSoup as soup

### Sample Penggunaan JSON dengan output Parameter Kabupaten

In [80]:
session = Session()

    # HEAD requests ask for *just* the headers, which is all you need to grab the
    # session cookie
session.head('https://bansm.kemdikbud.go.id/',verify=False)
#     session.cert=certifi.where()
response = session.post(
        url='https://bansm.kemdikbud.go.id/akreditasi/getKabKot',
        data={
            'idprov':prov[1]
        },
        headers={
            'Referer': 'https://bansm.kemdikbud.go.id/akreditasi'
        }
    )


In [81]:
print(response.text)

<option value="null">-- Pilih Kab/Kota terlebih dahulu --</option><option value='5103'>KABUPATEN BADUNG</option><option value='5106'>KABUPATEN BANGLI</option><option value='5108'>KABUPATEN BULELENG</option><option value='5104'>KABUPATEN GIANYAR</option><option value='5101'>KABUPATEN JEMBRANA</option><option value='5107'>KABUPATEN KARANG ASEM</option><option value='5105'>KABUPATEN KLUNGKUNG</option><option value='5102'>KABUPATEN TABANAN</option><option value='5171'>KOTA DENPASAR</option>


### Mendapatkan Semua id kabupaten dari tiap Provinsi

In [82]:

# from json import loads
# import certifi
# from collections import deque
all_kabid_array =[]
all_kabname_array =[]
for xx in range(len(prov)):
    print(prov[xx])

    # urlopen(request, context=ssl.create_default_context(cafile=certifi.where()))
    session = Session()

    # HEAD requests ask for *just* the headers, which is all you need to grab the
    # session cookie
    session.head('https://bansm.kemdikbud.go.id/',verify=False)
#     session.cert=certifi.where()
    response = session.post(
        url='https://bansm.kemdikbud.go.id/akreditasi/getKabKot',
        data={
            'idprov':prov[xx]
        },
        headers={
            'Referer': 'https://bansm.kemdikbud.go.id/akreditasi'
        }
    )
    # dic = loads(response.text)
    # print(response.json)
    # pd.DataFrame.from_dict(response.json)
    output = ""+response.text+""
    output = "<select>"+output+"</select>" 
    web_soup = soup(output)
    reviews = web_soup.find('select')
    delimiter = str(web_soup.contents[0]).split('"')
    id_kab = []
    for x in range(len(str(web_soup.contents[0]).split('"'))):
        if x%2 == 1:
            print(prov[xx]+'==')
            print(delimiter[x])
            id_kab.append(delimiter[x])
        
    all_kabid_array.append(id_kab[1:])
    nama_kabkot = []
    for value in reviews.stripped_strings:
        nama_kabkot.append(value)
        print(value)
    all_kabname_array.append(nama_kabkot[1:])
#         print(delimiter[x])


11
11==
null
11==
1107
11==
1112
11==
1108
11==
1116
11==
1103
11==
1102
11==
1114
11==
1106
11==
1104
11==
1105
11==
1111
11==
1117
11==
1110
11==
1113
11==
1115
11==
1109
11==
1118
11==
1101
11==
1171
11==
1173
11==
1174
11==
1172
11==
1175
-- Pilih Kab/Kota terlebih dahulu --
KABUPATEN ACEH BARAT
KABUPATEN ACEH BARAT DAYA
KABUPATEN ACEH BESAR
KABUPATEN ACEH JAYA
KABUPATEN ACEH SELATAN
KABUPATEN ACEH SINGKIL
KABUPATEN ACEH TAMIANG
KABUPATEN ACEH TENGAH
KABUPATEN ACEH TENGGARA
KABUPATEN ACEH TIMUR
KABUPATEN ACEH UTARA
KABUPATEN BENER MERIAH
KABUPATEN BIREUEN
KABUPATEN GAYO LUES
KABUPATEN NAGAN RAYA
KABUPATEN PIDIE
KABUPATEN PIDIE JAYA
KABUPATEN SIMEULUE
KOTA BANDA ACEH
KOTA LANGSA
KOTA LHOKSEUMAWE
KOTA SABANG
KOTA SUBULUSSALAM
51
51==
null
51==
5103
51==
5106
51==
5108
51==
5104
51==
5101
51==
5107
51==
5105
51==
5102
51==
5171
-- Pilih Kab/Kota terlebih dahulu --
KABUPATEN BADUNG
KABUPATEN BANGLI
KABUPATEN BULELENG
KABUPATEN GIANYAR
KABUPATEN JEMBRANA
KABUPATEN KARANG ASEM
KABUPATEN 

In [27]:
all_kabname_array

[['KABUPATEN ACEH BARAT',
  'KABUPATEN ACEH BARAT DAYA',
  'KABUPATEN ACEH BESAR',
  'KABUPATEN ACEH JAYA',
  'KABUPATEN ACEH SELATAN',
  'KABUPATEN ACEH SINGKIL',
  'KABUPATEN ACEH TAMIANG',
  'KABUPATEN ACEH TENGAH',
  'KABUPATEN ACEH TENGGARA',
  'KABUPATEN ACEH TIMUR',
  'KABUPATEN ACEH UTARA',
  'KABUPATEN BENER MERIAH',
  'KABUPATEN BIREUEN',
  'KABUPATEN GAYO LUES',
  'KABUPATEN NAGAN RAYA',
  'KABUPATEN PIDIE',
  'KABUPATEN PIDIE JAYA',
  'KABUPATEN SIMEULUE',
  'KOTA BANDA ACEH',
  'KOTA LANGSA',
  'KOTA LHOKSEUMAWE',
  'KOTA SABANG',
  'KOTA SUBULUSSALAM'],
 ['KABUPATEN BADUNG',
  'KABUPATEN BANGLI',
  'KABUPATEN BULELENG',
  'KABUPATEN GIANYAR',
  'KABUPATEN JEMBRANA',
  'KABUPATEN KARANG ASEM',
  'KABUPATEN KLUNGKUNG',
  'KABUPATEN TABANAN',
  'KOTA DENPASAR'],
 ['KABUPATEN LEBAK',
  'KABUPATEN PANDEGLANG',
  'KABUPATEN SERANG',
  'KABUPATEN TANGERANG',
  'KOTA CILEGON',
  'KOTA SERANG',
  'KOTA TANGERANG',
  'KOTA TANGERANG SELATAN'],
 ['KABUPATEN BENGKULU SELATAN',
  'KAB

### Menyatukan semua array menjadi satu dict

In [29]:
array = []
for x in range(len(all_kabid_array)):
    print(str(x)+'|'+prov[x]+'|'+nama_prov[x])
    for xx in range(len(all_kabid_array[x])):
        for xxx in range(len(jenjang)):

#             print(all_kabid_array[x][xx])
#             print(all_kabname_array[x][xx])
            dict_all = {
                'no': x,
                'id_prov':prov[x],
                'nama_prov':nama_prov[x],
                 'id_kabupaten':all_kabid_array[x][xx],
                'nama_kabupaten':all_kabname_array[x][xx],
                 'id_jenjang':jenjang[xxx],
                'nama_jenjang':nama_jenjang[xxx],
            }
            array.append(dict_all)
        

0|11|ACEH
1|51|BALI
2|36|BANTEN
3|17|BENGKULU
4|34|D.I. YOGYAKARTA
5|31|D.K.I. JAKARTA
6|75|GORONTALO
7|15|JAMBI
8|32|JAWA BARAT
9|33|JAWA TENGAH
10|35|JAWA TIMUR
11|61|KALIMANTAN BARAT
12|63|KALIMANTAN SELATAN
13|62|KALIMANTAN TENGAH
14|64|KALIMANTAN TIMUR
15|65|KALIMANTAN UTARA
16|19|KEPULAUAN BANGKA BELITUNG
17|21|KEPULAUAN RIAU
18|18|LAMPUNG
19|9999|LUAR NEGERI
20|81|MALUKU
21|82|MALUKU UTARA
22|52|NUSA TENGGARA BARAT
23|53|NUSA TENGGARA TIMUR
24|94|PAPUA
25|91|PAPUA BARAT
26|14|RIAU
27|76|SULAWESI BARAT
28|73|SULAWESI SELATAN
29|72|SULAWESI TENGAH
30|74|SULAWESI TENGGARA
31|71|SULAWESI UTARA
32|13|SUMATERA BARAT
33|16|SUMATERA SELATAN
34|12|SUMATERA UTARA


In [32]:
array

[{'no': 0,
  'id_prov': '11',
  'nama_prov': 'ACEH',
  'id_kabupaten': '1107',
  'nama_kabupaten': 'KABUPATEN ACEH BARAT',
  'id_jenjang': '1',
  'nama_jenjang': 'SD/MI'},
 {'no': 0,
  'id_prov': '11',
  'nama_prov': 'ACEH',
  'id_kabupaten': '1107',
  'nama_kabupaten': 'KABUPATEN ACEH BARAT',
  'id_jenjang': '2',
  'nama_jenjang': 'SMP/MTS'},
 {'no': 0,
  'id_prov': '11',
  'nama_prov': 'ACEH',
  'id_kabupaten': '1107',
  'nama_kabupaten': 'KABUPATEN ACEH BARAT',
  'id_jenjang': '3',
  'nama_jenjang': 'SMK'},
 {'no': 0,
  'id_prov': '11',
  'nama_prov': 'ACEH',
  'id_kabupaten': '1107',
  'nama_kabupaten': 'KABUPATEN ACEH BARAT',
  'id_jenjang': '4',
  'nama_jenjang': 'SMA/MA'},
 {'no': 0,
  'id_prov': '11',
  'nama_prov': 'ACEH',
  'id_kabupaten': '1107',
  'nama_kabupaten': 'KABUPATEN ACEH BARAT',
  'id_jenjang': '5',
  'nama_jenjang': 'SLB'},
 {'no': 0,
  'id_prov': '11',
  'nama_prov': 'ACEH',
  'id_kabupaten': '1107',
  'nama_kabupaten': 'KABUPATEN ACEH BARAT',
  'id_jenjang': '6

### Merubah Menjadi Dataframe

In [33]:
prov_kab = pd.DataFrame(array)

In [34]:
prov_kab

no id_prov       nama_prov id_kabupaten        nama_kabupaten  \
0      0      11            ACEH         1107  KABUPATEN ACEH BARAT   
1      0      11            ACEH         1107  KABUPATEN ACEH BARAT   
2      0      11            ACEH         1107  KABUPATEN ACEH BARAT   
3      0      11            ACEH         1107  KABUPATEN ACEH BARAT   
4      0      11            ACEH         1107  KABUPATEN ACEH BARAT   
...   ..     ...             ...          ...                   ...   
5781  34      12  SUMATERA UTARA         1274    KOTA TEBING TINGGI   
5782  34      12  SUMATERA UTARA         1274    KOTA TEBING TINGGI   
5783  34      12  SUMATERA UTARA         1274    KOTA TEBING TINGGI   
5784  34      12  SUMATERA UTARA         1274    KOTA TEBING TINGGI   
5785  34      12  SUMATERA UTARA         1274    KOTA TEBING TINGGI   

     id_jenjang nama_jenjang  
0             1        SD/MI  
1             2      SMP/MTS  
2             3          SMK  
3             4       SMA/MA  
4             5          SLB  
...         ...          ...  
5781          7        SMPLB  
5782          8         SMLB  
5783          9       SPK SD  
5784         10      SPK SMP  
5785         11      SPK SMA  

[5786 rows x 7 columns]

# Mendapatkan Data

### Sample Mendapatkan Data pada 1 kabupaten dan 1 Jenjang

In [48]:
id_provinsi = prov_kab['id_prov'][0]
id_kabupaten = prov_kab['id_kabupaten'][0]
id_jenjang = prov_kab['id_jenjang'][0]

In [49]:
print("ID PROVINSI    : "+str(id_provinsi)+' |'+prov_kab['nama_prov'][0])
print("ID KABUPATEN   : "+str(id_kabupaten)+' |'+prov_kab['nama_kabupaten'][0])
print("ID JENJANG     : "+str(id_jenjang)+' |'+prov_kab['nama_jenjang'][0])


ID PROVINSI    : 11 |ACEH
ID KABUPATEN   : 1107 |KABUPATEN ACEH BARAT
ID JENJANG     : 1 |SD/MI


In [50]:
session.head('https://bansm.kemdikbud.go.id/',verify=False)
# session.cert=certifi.where()
response = session.post(
    url='https://bansm.kemdikbud.go.id/akreditasi/search/0',
    data={'page':'0',
          'keywords':'',
         'prov':prov_kab['id_prov'][0],
         'kabkot':prov_kab['id_kabupaten'][0],
          'jenj':prov_kab['id_jenjang'][0]
         ,}
    ,
    headers={
        'Referer': 'https://bansm.kemdikbud.go.id/akreditasi'
    }
)

output = ""+response.text+""
output

c:\users\zahru\appdata\local\programs\python\python38-32\lib\site-packages\urllib3\connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'bansm.kemdikbud.go.id'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
c:\users\zahru\appdata\local\programs\python\python38-32\lib\site-packages\urllib3\connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'bansm.kemdikbud.go.id'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(


'<div class="md-padding" style="padding-top: 35px;padding-bottom: 30px;">\r\n  <div class="container">\r\n    <div class="col-md-12">\r\n      <div class="heading style4 sm" style="margin-bottom: 25px; ">\r\n        <h4 class="uppercase">Hasil Pencarian Data Sekolah</h4>\r\n      </div>\r\n\t  <div class=\'col-sm-12\'>\r\n\t  \t\t  <ul id="list" class="pagination pull-left">\r\n\t\t\t<li class="active"><a href="javascript:void(0)">272 Sekolah</a></li>&nbsp;&nbsp;\r\n\t\t</ul>\r\n\t  \t          <script>\r\n        function getData(page){  \r\n            $.ajax({\r\n                method: "POST",\r\n                url: "https://bansm.kemdikbud.go.id/akreditasi/search/"+page,\r\n                data: { page: page },\r\n                beforeSend: function(){\r\n                    $(\'.loading\').show();\r\n                },\r\n                success: function(data){\r\n                    $(\'.loading\').hide();\r\n                    $(\'#datana\').html(data);\r\n                }

### Melihat Jumlah Data Yang Ada

In [52]:
try:
    jumlah = int(output.split('<li class="active"><a href="javascript:void(0)">')[1].split('</a></li>&nbsp;&nbsp;\r\n\t\t</ul>')[0].split()[0])
    print(jumlah)
except:
    jumlah = 0

272


Jumlah Data 272 digunakan untuk mendapatkan page,dalam website ada 272 data yang tiap pagenya ada 10 data,
yang berarti ada 28 Page

### Mendapatkan Semua Data semua Page 

In [53]:
data = []
if jumlah%10 > 0:
    loop = int(jumlah/10)+1
else :
    loop = int(jumlah/10)
    
for x in range(loop):
    session.head('https://bansm.kemdikbud.go.id/',verify=False)
#     session.cert=certifi.where()
    response = session.post(
        url='https://bansm.kemdikbud.go.id/akreditasi/search/0',
        data={'page':x*10,
              'keywords':'',
             'prov':prov_kab['id_prov'][0],
             'kabkot':prov_kab['id_kabupaten'][0],
          'jenj':prov_kab['id_jenjang'][0]}
        ,
        headers={
            'Referer': 'https://bansm.kemdikbud.go.id/akreditasi'
        }
    )

    output = ""+response.text+""
    data.append(pd.read_html(output)[0])
#     print(x)

c:\users\zahru\appdata\local\programs\python\python38-32\lib\site-packages\urllib3\connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'bansm.kemdikbud.go.id'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
c:\users\zahru\appdata\local\programs\python\python38-32\lib\site-packages\urllib3\connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'bansm.kemdikbud.go.id'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
c:\users\zahru\appdata\local\programs\python\python38-32\lib\site-packages\urllib3\connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'bansm.kemdikbud.go.id'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/late

### Menggabungkan Setiap Page

In [56]:
final = pd.concat(data)

In [59]:
final

No      NPSN                      Sekolah  Tahun Akreditasi Peringkat  \
0     1  10102473                 SMPN 1 WOYLA              2015         B   
1     2  10102476              SMPN 2 MEULABOH              2015         A   
2     3  10102482           SMPN 1 WOYLA BARAT              2015         B   
3     4  10102488       SMPN 2 PANTE CEUREUMEN              2013         B   
4     5  10102497              SMPN 1 MEULABOH              2015         A   
..  ...       ...                          ...               ...       ...   
7   268  69786607               SDS LUENG BARO              2015         A   
8   269  69829370  SMP SATU ATAP BLANG TEUNGOH              2013         B   
9   270  69900191            SMP N 7 KAWAY XVI              2014         A   
0   271  69943919    SMP SATU ATAP KUBU CAPANG              2015         A   
1   272  69950910          SMP N 4 WOYLA BARAT              2014         B   

   Alamat  
0     NaN  
1     NaN  
2     NaN  
3     NaN  
4     NaN  
..    ...  
7     NaN  
8     NaN  
9     NaN  
0     NaN  
1     NaN  

[272 rows x 6 columns]

In [67]:
final_all = []
for xxx in range(len(prov_kab)):
    print("---------------->  "+str(xxx)+'/'+str(len(prov_kab)))
    print("----------->  "+prov_kab['nama_prov'][xxx])
    print("------>  "+prov_kab['nama_kabupaten'][xxx])
    print("------>  "+prov_kab['nama_jenjang'][xxx])
    session.head('https://bansm.kemdikbud.go.id/',verify=False)
#     session.cert=certifi.where()
    response = session.post(
        url='https://bansm.kemdikbud.go.id/akreditasi/search/0',
        data={'page':'0',
              'keywords':'',
             'prov':prov_kab['id_prov'][xxx],
             'kabkot':prov_kab['id_kabupaten'][xxx],
              'jenj':prov_kab['id_jenjang'][xxx],}
        ,
        headers={
            'Referer': 'https://bansm.kemdikbud.go.id/akreditasi'
        }
    )

    output = ""+response.text+""
    output
    try:
        jumlah = int(output.split('<li class="active"><a href="javascript:void(0)">')[1].split('</a></li>&nbsp;&nbsp;\r\n\t\t</ul>')[0].split()[0])
#         print(jumlah)
    except:
        try:
            jumlah = int(output.split('Showing : ')[1].split('</div>')[0].split()[0])
#             print("Showing ==========================>>>"+str(jumlah))
        except:
            jumlah = 0
    data = []
    if jumlah == 0:
        continue
    if jumlah%10 > 0:
        loop = int(jumlah/10)+1
    else :
        loop = int(jumlah/10)

    for x in range(loop):
#         print(x)
        session.head('https://bansm.kemdikbud.go.id/',verify=False)
#         session.cert=certifi.where()
        response = session.post(
            url='https://bansm.kemdikbud.go.id/akreditasi/search/0',
            data={'page':x*10,
                  'keywords':'',
                 'prov':prov_kab['id_prov'][xxx],
                 'kabkot':prov_kab['id_kabupaten'][xxx],
                  'jenj':prov_kab['id_jenjang'][xxx],}
            ,
            headers={
                'Referer': 'https://bansm.kemdikbud.go.id/akreditasi'
            }
        )

        output = ""+response.text+""
        data.append(pd.read_html(output)[0])
    final = pd.concat(data)
    final['Provinsi'] = prov_kab['nama_prov'][xxx]
    final['Kabupaten'] = prov_kab['nama_kabupaten'][xxx]
    final['Jenjang'] = prov_kab['nama_jenjang'][xxx]
    final_all.append(final)
    #     print(x)

---------------->  0/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SD/MI
---------------->  1/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SMP/MTS
---------------->  2/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SMK
---------------->  3/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SMA/MA
---------------->  4/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SLB
---------------->  5/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SDLB
---------------->  6/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SMPLB
---------------->  7/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SMLB
---------------->  8/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SPK SD
---------------->  9/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SPK SMP
---------------->  10/5786
----------->  ACEH
------>  KABUPATEN ACEH BARAT
------>  SPK SMA
-------

KeyboardInterrupt: 

In [68]:
super_final_all = pd.concat(final_all)

In [69]:
super_final_all

No      NPSN                     Sekolah  Tahun Akreditasi Peringkat  \
0    1  10102473                SMPN 1 WOYLA              2015         B   
1    2  10102476             SMPN 2 MEULABOH              2015         A   
2    3  10102482          SMPN 1 WOYLA BARAT              2015         B   
3    4  10102488      SMPN 2 PANTE CEUREUMEN              2013         B   
4    5  10102497             SMPN 1 MEULABOH              2015         A   
..  ..       ...                         ...               ...       ...   
4   15  60714717          MIS SUNAN KALIJOGO              2008         B   
0    1  20549288  SLB MARSUDI UTOMO KESAMBEN              2019         B   
1    2  20549314          SLB - B NGUDI HAYU              2019         B   
2    3  69899994            SLB TUNAS BANGSA              2019         C   
3    4  69954628         SLB PGRI MUJI UTOMO              2019         C   

                                      Alamat    Provinsi  \
0                                        NaN        ACEH   
1                                        NaN        ACEH   
2                                        NaN        ACEH   
3                                        NaN        ACEH   
4                                        NaN        ACEH   
..                                       ...         ...   
4                                        NaN  JAWA TIMUR   
0                                   KESAMBEN  JAWA TIMUR   
1   JL. RAYATOGOGAN 001B LINGK. KENDALDOYONG  JAWA TIMUR   
2                     JL. PONDOK KIDUL NO. 3  JAWA TIMUR   
3             DSN. MANDINGASEM RT. 02 RW. 01  JAWA TIMUR   

               Kabupaten Jenjang  
0   KABUPATEN ACEH BARAT   SD/MI  
1   KABUPATEN ACEH BARAT   SD/MI  
2   KABUPATEN ACEH BARAT   SD/MI  
3   KABUPATEN ACEH BARAT   SD/MI  
4   KABUPATEN ACEH BARAT   SD/MI  
..                   ...     ...  
4       KABUPATEN BLITAR  SMA/MA  
0       KABUPATEN BLITAR     SLB  
1       KABUPATEN BLITAR     SLB  
2       KABUPATEN BLITAR     SLB  
3       KABUPATEN BLITAR     SLB  

[107508 rows x 9 columns]